backend/content-type/eCampMultiSelect/src/Strategy.php

```diff
@@ -32,7 +32,7 @@ public function contentNodeExtract(ContentNode $contentNode): array {
      * @throws ORMException
      * @throws EntityNotFoundException
      */
-    public function contentNodeCreated(ContentNode $contentNode): void {
+    public function contentNodeCreated(ContentNode $contentNode, ?ContentNode $prototype = null): void {
         foreach ($contentNode->getContentType()->getJsonConfig()['items'] as $key => $configItem) {
             $option = $this->optionService->createEntity(['pos' => $key, 'translateKey' => $configItem, 'checked' => false], $contentNode);
             $this->getServiceUtils()->emPersist($option);
```

Comment:
Ich sehe eher dass Inkonsistenzen auftreten können, wenn du hier nichts vom Prototype kopierst. Wenn wir die Options in Zukunft mal anpassen müssen wir ja auch entweder rückwärtskompatibel sein oder alle bestehenden MultiSelects migrieren oder? Sonst könnte man die alten MultiSelects (wie auch der Prototype eines ist) nicht mehr im UI rendern. Somit müsste ein Prototype hier eigentlich immer risikofrei klonbar sein.

Reply:
Das dürfte glaubs nicht das Problem sein. Die Options werden ja in Zeile 36-38 bei jedem Erstellen vollkopiert (einfach nicht vom ProtoType sondern vom ursprünglichen ContentType).

Ich denke aber, dass `checked` vom prototype kopiert werden müsste anstatt per default auf `false` gesetzt.

Reply:
Das mit dem Checked habe ich mir auch überlegt...
Aber genau da kann es schwer werden, wenn in der Config und in den Detail-Daten (Options) nicht zusammenpassen...

Reply:
Daher meinte ich eben, dass falls ein Prototype vorhanden ist die Options nicht aus der ContentType JSONConfig erstellt werden sollten, sondern als Klone der bestehenden Optionen des Prototypes. Gleich inklusive `checked` Status. Übersehe ich dabei etwas?

Reply:
Ich kann mit dem Vorschlag leben.
Es gibt aber auch hier eine "Lücke".

Wenn wir die Config anpassen wollen, bekommt der Benutzer zwei verschiedene "MultiSelect" Versionen.
Beim erstellen des Blocks jene von der Config; und wenn er im Block spontan eine neue hinzufügt, jene von der Config.
Könnte halt auch verwirrend sein.

Bin unsicher, ob es hier überhaupt ein "richtig" gibt.

Reply:
Ja, darum meinte ich eben, wenn wir die Config anpassen müssen haben wir zwei Optionen: Entweder wir machen nur eine rückwärtskompatible Änderung und nehmen diese mögliche Diskrepanz in Kauf, oder wir migrieren gleichzeitig alle bestehenden MultiSelects, sodass sie mit der neuen Config konform sind.

Reply:
OK.
Ich hab's mal angepasst. ContentType MultiSelect erstellt nun eine tiefe Kopie, falls ein Prototype angegeben wird.
(Also bei Activity erstellen).